tests/integration-tests/tests/cloudwatch_logging/test_cloudwatch_logging.py

```diff
@@ -151,31 +172,116 @@ def _read_log_configs_from_master(self):
         config = json.loads(self._run_command_on_master(read_cmd))
         return config.get("log_configs")
 
+    def _read_master_node_config(self):
+        """Read the node configuration JSON file at NODE_CONFIG_PATH on the master node."""
+        read_cmd = "cat {0}".format(NODE_CONFIG_PATH)
+        master_node_config = json.loads(self._run_command_on_master(read_cmd)).get("cfncluster", {})
+        assert_that(master_node_config).is_not_empty()
+        LOGGER.info("DNA config read from master node: {0}".format(_dump_json(master_node_config)))
+        return master_node_config
+
+    def _read_compute_node_config(self):
+        """Read the node configuration JSON file at NODE_CONFIG_PATH on a compute node."""
+        compute_node_config = {}
+        compute_hostname_to_config = self._run_command_on_computes("cat {{redirect}} {0}".format(NODE_CONFIG_PATH))
+
+        # Use first one, since ParallelCluster-specific node config should be the same on every compute node
+        for _, config_json in compute_hostname_to_config.items():
+            compute_node_config = json.loads(config_json).get("cfncluster", {})
+            break
+
+        assert_that(compute_node_config).is_not_empty()
+        LOGGER.info("DNA config read from compute node: {0}".format(_dump_json(compute_node_config)))
+        return compute_node_config
+
+    def _read_node_configs(self):
+        """Return a dict mapping node role names to the config at NODE_CONFIG_PATH."""
+        return {
+            MASTER_NODE_ROLE_NAME: self._read_master_node_config(),
+            COMPUTE_NODE_ROLE_NAME: self._read_compute_node_config(),
+        }
+
     @staticmethod
     def _clean_log_config(log):
         """Remove unnecessary fields from the given log dict."""
         desired_keys = ["file_path", "log_stream_name", "feature_conditions"]
         return {key: log[key] for key in desired_keys}
 
-    def _filter_logs(self, logs):
-        """Populate self._relevant_logs with logs appropriate for the two different node types."""
+    def _filter_logs_on_platform_and_scheduler(self, logs):
+        """Filter from logs all entries that don't support the cluster's OS or scheduler."""
+        return [
+            log for log in logs if self.scheduler in log.get("schedulers") and self.platform in log.get("platforms")
+        ]
+
+    def _log_is_relevant_for_feature(self, log, node_config):
+        """Return a boolean describing whether log contains a feature_conditions entry relevant to self.feature_key."""
+        is_relevant_for_feature = False
+        for feature_condition in log.get("feature_conditions", []):
+            if all(
+                [
+                    self.feature_key == feature_condition.get("dna_key"),
+                    node_config.get(self.feature_key) in feature_condition.get("satisfying_values"),
+                ]
+            ):
+                is_relevant_for_feature = True
```

Comment:
Minor: no need to use the variable `is_relevant_for_feature` and `break`. `return True` and a `return False` at the end of the function should function the same.

Reply:
Yeah the thought was to have only one exit point for the function. I'll change it, though.